wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraConnectionDestroyerPlugin.java

```diff
@@ -0,0 +1,161 @@
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Logger;
+import software.amazon.jdbc.HostSpec;
+import software.amazon.jdbc.JdbcCallable;
+import software.amazon.jdbc.NodeChangeOptions;
+import software.amazon.jdbc.OpenedConnectionTracker;
+import software.amazon.jdbc.PluginService;
+import software.amazon.jdbc.hostlistprovider.AuroraHostListProvider;
+import software.amazon.jdbc.plugin.failover.FailoverSQLException;
+import software.amazon.jdbc.util.RdsUtils;
+import software.amazon.jdbc.util.StringUtils;
+import software.amazon.jdbc.util.SubscribedMethodHelper;
+
+public class AuroraConnectionDestroyerPlugin extends AbstractConnectionPlugin {
+
+  static final String MYSQL_GET_INSTANCE_NAME_SQL = "SELECT @@aurora_server_id";
+  static final String MYSQL_GET_INSTANCE_NAME_COL = "@@aurora_server_id";
+  static final String PG_GET_INSTANCE_NAME_SQL = "SELECT aurora_db_instance_identifier()";
+  static final String PG_INSTANCE_NAME_COL = "aurora_db_instance_identifier";
+  static final String METHOD_ABORT = "abort";
+  static final String METHOD_CLOSE = "close";
+  private static final String PG_DRIVER_PROTOCOL = "postgresql";
+  private static final Set<String> subscribedMethods =
+      Collections.unmodifiableSet(new HashSet<String>() {
+        {
+          addAll(SubscribedMethodHelper.NETWORK_BOUND_METHODS);
+          add("connect");
+          add("notifyNodeListChanged");
+        }
+      });
+
+  private final PluginService pluginService;
+  private final Properties props;
+  private final RdsUtils rdsHelper;
+  private String retrieveInstanceQuery;
+  private String instanceNameCol;
+  private String clusterInstanceTemplate;
+
+  AuroraConnectionDestroyerPlugin(PluginService pluginService, Properties props) {
+    this.pluginService = pluginService;
+    this.props = props;
+    this.rdsHelper = new RdsUtils();
+  }
+
+  @Override
+  public Set<String> getSubscribedMethods() {
+    return subscribedMethods;
+  }
+
+  @Override
+  public Connection connect(String driverProtocol, HostSpec hostSpec, Properties props,
+      boolean isInitialConnection, JdbcCallable<Connection, SQLException> connectFunc) throws SQLException {
+    if (driverProtocol.contains(PG_DRIVER_PROTOCOL)) {
+      this.retrieveInstanceQuery = PG_GET_INSTANCE_NAME_SQL;
+      this.instanceNameCol = PG_INSTANCE_NAME_COL;
+    } else {
+      this.retrieveInstanceQuery = MYSQL_GET_INSTANCE_NAME_SQL;
+      this.instanceNameCol = MYSQL_GET_INSTANCE_NAME_COL;
+    }
+
+    final Connection conn = connectFunc.call();
+
+    if (conn != null) {
+      final HostSpec currentHostSpec = this.pluginService.getCurrentHostSpec();
+      if (rdsHelper.isRdsClusterDns(currentHostSpec.getHost())) {
+        currentHostSpec.addAlias(getInstanceEndpoint(conn, currentHostSpec));
+      }
+    }
+
+    OpenedConnectionTracker.populateOpenedConnectionQueue(hostSpec, conn);
```

Comment:
Can we make populateOpenedConnectionQueue() not static? It'd allow us to pass a reference of OpenedConnectionTracker as a plugin dependency and simplify unit testing. Static dependencies are hard to mock.